Assets/MRTK/SDK/Features/Input/Handlers/Constraints/ConstraintManager.cs

```diff
@@ -59,5 +65,25 @@ private void ApplyConstraintsForType(ref MixedRealityTransform transform, bool i
                 }
             }
         }
+
+        private void EnsureNewConstraintsInitialized()
+        {
+            var currentConstraints = host.GetComponents<TransformConstraint>();
```

Comment:
Do you know what the calling mechanics are of EnsureNewConstraintsInitialized() - i.e. potentially on every single frame update?

Because if so, then the costs over time will probably not be that good - we'll end up doing component find/enumerate and then run a non-hashed array find on every single frame.

I think that we had a similar situation with some tap to place or solver stuff, where instead the constraints, when they were added to an object, would look for the SolverManager or something and be like "yo, I"m here now". You'd have to also make sure that the SolverManager/ConstraintManager would need to do a one-time enumeration when it started up, so that it could catch cases where individual constraints/solvers were added before the manager got added.